### ISSUE TITLE
Throwing Error when coupled with limit fileSize of multer

### ISSUE THREAD

**Issue by LorenzoElenkov** [khaosdoctor/multer-firebase-storage#4]
TL;DR
I am getting `ApiError: The specified key does not exist.` when file is larger than what is specified in multer configs.

```
const Multer = require('multer');
const firebaseStorage = require('multer-firebase-storage');
const fbAdmin = require('firebase-admin');
const serviceAccount = require('../serviceAccount.json');

const multer = Multer({
    storage: firebaseStorage({
        bucketName: process.env.FB_STORAGEBUCKET,
        credentials: fbAdmin.credential.cert(serviceAccount),
        public: true,
        hooks: {
            beforeUpload: (req, file) => {
                file.originalname = 'dsds';
            }
        }
    }),
    limits: {
        fileSize: 2 * 1024 * 1024
    },
});
```

Then in my route:

`router.post('/upload', multer.single('photo'), async (req, res) => {
    //doing something else
})`

Also I have this error handling in my server.js

`app.use((err, req, res, next) => {
    if (err.code === "LIMIT_FILE_SIZE") {
        return res.send({ success: false , error: { code: 413, message: 'File is too big'}});
    }
})`

The response from the error handler for LIMIT_FILE_SIZE is sent back to the front end but it throws this error on the server side and ***CRASHES*** the server

```
ApiError: The specified key does not exist.
    at new ApiError (C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\@google-cloud\storage\build\src\nodejs-common\util.js:78:15)
    at Util.parseHttpRespBody (C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\@google-cloud\storage\build\src\nodejs-common\util.js:213:38)
    at Util.handleResp (C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\@google-cloud\storage\build\src\nodejs-common\util.js:154:117)
    at C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\@google-cloud\storage\build\src\nodejs-common\util.js:500:22
    at onResponse (C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\retry-request\index.js:228:7)
    at C:\Users\LorenzoElenkov\Desktop\Coding_tests\Assignments\1_Instagram\node_modules\multer-firebase-storage\node_modules\teeny-request\build\src\index.js:226:13
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 404,
  errors: [
    {
      message: 'The specified key does not exist.',
      domain: 'global',
      reason: 'notFound'
    }
  ],
  response: PassThrough {
    _readableState: ReadableState {
      objectMode: false,
      highWaterMark: 16384,
      buffer: BufferList { head: null, tail: null, length: 0 },
      length: 0,
      pipes: [],
      flowing: true,
      ended: true,
      endEmitted: true,
      reading: false,
      constructed: true,
      sync: false,
      needReadable: false,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      destroyed: true,
      errored: null,
      closed: true,
      closeEmitted: true,
      defaultEncoding: 'utf8',
      awaitDrainWriters: null,
      multiAwaitDrain: false,
      readingMore: false,
      dataEmitted: true,
      decoder: null,
      encoding: null,
      [Symbol(kPaused)]: false
    },
    _events: [Object: null prototype] {
      prefinish: [Function: prefinish],
      error: [ [Function (anonymous)], [Function (anonymous)] ],
      data: [Function (anonymous)],
      end: [Function (anonymous)]
    },
    _eventsCount: 4,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: true,
      needDrain: false,
      ending: true,
      ended: true,
      finished: true,
      destroyed: true,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: true,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: true,
      closeEmitted: true,
      [Symbol(kOnFinished)]: []
    },
    allowHalfOpen: true,
    statusCode: 404,
    statusMessage: 'Not Found',
    request: {
      agent: Agent {
        _events: [Object: null prototype] {
          free: [Function (anonymous)],
          newListener: [Function: maybeEnableKeylog]
        },
        _eventsCount: 2,
        _maxListeners: undefined,
        defaultPort: 443,
        protocol: 'https:',
        options: [Object: null prototype] {
          maxSockets: Infinity,
          keepAlive: true,
          path: null
        },
        requests: [Object: null prototype] {},
        sockets: [Object: null prototype] {},
        freeSockets: [Object: null prototype] {
          'storage.googleapis.com:443:::::::::::::::::::::': [ [TLSSocket], [TLSSocket] ]
        },
        keepAliveMsecs: 1000,
        keepAlive: true,
        maxSockets: Infinity,
        maxFreeSockets: 256,
        scheduling: 'lifo',
        maxTotalSockets: Infinity,
        totalSocketCount: 2,
        maxCachedSessions: 100,
        _sessionCache: {
          map: {
            'storage.googleapis.com:443:::::::::::::::::::::': [Buffer [Uint8Array]]
          },
          list: [ 'storage.googleapis.com:443:::::::::::::::::::::' ]
        },
        [Symbol(kCapture)]: false
      },
      headers: {
        'User-Agent': 'gcloud-node-storage/5.20.5',
        'x-goog-api-client': 'gl-node/16.16.0 gccl/5.20.5 gccl-invocation-id/94b60a66-e9d0-41bf-acbc-885236d00ef8',
        Authorization: 'Bearer ya29.c.b0AUFJQsEKnLCLBrKpucYvOnipwvy_4IdGZLtpU4_UqXMiFnyRq0Kasa4KNPjMUJt6WTYcTNmAii6IHTtrJNXQHM3_Cg-XdLgNg-6UG6vnCXiUCGVdFbr_onA42K-7PcnmQ3-EcraKbzQ79H7WuoBGf4HIrdTBKT73YL1sNoOy-lq4lVCOe9DNYMxkLMm42xAsKVEK8km3YC0Apyl_rwXwOJx8Cl25LlUabMWE...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................',
        'Content-Type': 'application/json'
      },
      href: 'https://storage.googleapis.com/storage/v1/b/training-instagram-coherent.appspot.com/o/1664476539476--photo./acl?'
    },
    body: '{\n' +
      '  "error": {\n' +
      '    "code": 404,\n' +
      '    "message": "The specified key does not exist.",\n' +
      '    "errors": [\n' +
      '      {\n' +
      '        "message": "The specified key does not exist.",\n' +
      '        "domain": "global",\n' +
      '        "reason": "notFound"\n' +
      '      }\n' +
      '    ]\n' +
      '  }\n' +
      '}\n',
    headers: {
      'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"',
      'cache-control': 'no-cache, no-store, max-age=0, must-revalidate',
      'content-length': '241',
      'content-type': 'application/json; charset=UTF-8',
      date: 'Thu, 29 Sep 2022 18:35:40 GMT',
      expires: 'Mon, 01 Jan 1990 00:00:00 GMT',
      pragma: 'no-cache',
      server: 'UploadServer',
      vary: 'Origin, X-Origin',
      'x-guploader-uploadid': 'ADPycdsD04BVQI5fJJrI0qrDDoXqvD6WJ9ioG_CePCxpUjbeyNfTi39b_4bXe7j-yEad6lnWkWo1tt-tEOb2fRa4rDHrT16BK485'
    },
    toJSON: [Function: toJSON],
    [Symbol(kCapture)]: false,
    [Symbol(kCallback)]: null
  }
}
```

**Comment by LorenzoElenkov** [comment]
It seems that multer bypasses the fileSize limitation configuration and then it proceeds to upload it to firebase storage, in my case, which throws an error on its end.

As a remedy for my case - I came across another workaround checking the length of the file:
***THIS WORKS BEST IF ONLY A SINGLE FILE IS SENT TO THE SERVER***
```
router.post('/upload', async (req, res) => {
    const size = req.headers['content-length'] / 1024 / 1024;
    if (size > 3) {
        return res.json({ ok: false, message: 'File exceeds the 3MB limit' })
    } else {
        upload.single('photo')(req, res, err => {
            if (err) {
                return res.json({ ok: false, message: err.message });
            } else {
                console.log(req.file?.publicUrl);
            }
        })
    }
})
```
and `limits: { fileSize: 3 * 1024 * 1024 }` from multer config object is removed.

I hope this helps someone!

**Comment by khaosdoctor** [comment]
Hey! Thanks for the comments, I was on vacation but I'm back and I'll take a look and see if I can reproduce it locally!
